### ISSUE TITLE
Certain DNG files cannot be processed

### ISSUE THREAD

**Issue by KingOfDog** [immich-app/immich#13029]
### The bug

Certain DNG files of mine can't be processed by Immich. They only show up as broken images, while the metadata is correctly displayed. It appears to mostly occur with DNG files that we're created by Lightroom, e.g. through HDR or Panorama stitching, of ARW raw photos from my Sony a7 IV camera. See the example image [here](https://cloud.marcelstruck.de/s/PnDEzjAwCoHLd8Q).

![image](https://github.com/user-attachments/assets/d4dd2a25-e5f6-409e-9c8e-d1051fcf138f)

I've tracked down the issue to the `libraw` library not being able to parse the DNG files. If I build `libraw` from source with the Adobe DNG SDK linked, on the other hand, it is actually able to read out details from the file. `raw-identify` then says that the file is a `Sony ILCE-7M4` image.

Would it be possible to include the Adobe DNG SDK in Immich server in any way? Or is there an easier way to get those files to show up?

### The OS that Immich Server is running on

Unraid 6.12.10

### Version of Immich Server

1.116.2

### Version of Immich Mobile App

1.116.1 build.161

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.transcoding.yml
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /mnt/user/KingOfDog/Photos/:/mnt/photos
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine@sha256:d6c2911ac51b289db208767581a5d154544f2b2fe4914ea5056443f62dc6e900
    healthcheck:
      test: redis-cli ping || exit 1
    restart: always

  database:
    container_name: immich_postgres
    image: docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: '--data-checksums'
    volumes:
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT SUM(checksum_failures) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
      interval: 5m
      # start_interval: 30s
      start_period: 5m
    command: ["postgres", "-c" ,"shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors', "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
    restart: always
  
volumes:
  model-cache:
```

### Your .env content

```Shell
# The location where your uploaded files are stored
UPLOAD_LOCATION=/mnt/user/immich/upload
# The location where your database files are stored
DB_DATA_LOCATION=/mnt/user/immich/database

# To set a timezone, uncomment the next line and change Etc/UTC to a TZ identifier from this list: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
# TZ=Etc/UTC

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=*****

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```

### Reproduction steps

1. Upload the [file](https://cloud.marcelstruck.de/s/PnDEzjAwCoHLd8Q) to Immich
2. The image won't show up

### Relevant log output

```shell
[Nest] 7  - 09/29/2024, 10:32:17 AM   ERROR [Microservices:JobService] Unable to run job handler (thumbnailGeneration/generate-preview): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/photos/2024/2024-09/2024-09-19_Rückfahrt nach Trondheim/Sony a7 IV Photo/_A7F9420-HDR.dng' @ error/dng.c/ReadDNGImage/504 (null)
[Nest] 7  - 09/29/2024, 10:32:17 AM   ERROR [Microservices:JobService] Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/photos/2024/2024-09/2024-09-19_Rückfahrt nach Trondheim/Sony a7 IV Photo/_A7F9420-HDR.dng' @ error/dng.c/ReadDNGImage/504 (null)
    at Sharp.toFile (/usr/src/app/node_modules/sharp/lib/output.js:90:19)
    at MediaRepository.generateThumbnail (/usr/src/app/dist/repositories/media.repository.js:69:14)
    at MediaService.generateThumbnail (/usr/src/app/dist/services/media.service.js:174:48)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async MediaService.handleGeneratePreview (/usr/src/app/dist/services/media.service.js:139:29)
    at async /usr/src/app/dist/services/job.service.js:171:36
    at async Worker.processJob (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:394:28)
    at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
[Nest] 7  - 09/29/2024, 10:32:17 AM   ERROR [Microservices:JobService] Object:
{
  "id": "4c24f2af-dab2-4634-8236-2a46c01716c6"
}
```

**Comment by alexisharara** [comment]
Same issue with Apple's new JPEG-XL format on iOS 18 which stores them in a .DNG container

**Comment by mertalev** [comment]
> Would it be possible to include the Adobe DNG SDK in Immich server in any way?

We build libraw ourselves, so this is possible. IIRC it didn't have good support for Linux (only Windows and Mac), so it's a bit of a pain to get it working. A PR for the [base image](https://github.com/immich-app/base-images) would be welcome if you're able to make it work in Linux.

**Comment by MikevE1995** [comment]
Same issue on Galaxy 24 Ultra and Windows pc

**Comment by thomashollier** [comment]
Same here, using v1.117.

**Comment by devtechk** [comment]
Hopefullly I'm not alone... some dng are displayed well other not... like yours. I hope upgrade will adjust it

**Comment by jedie** [comment]
Same here with [v1.119.1](https://github.com/immich-app/immich/releases/tag/v1.119.1) and DNG from different camery types: Pentax K-1, Pentax K-5 II, Pentax K-01 and more ... All of them display only the very low res preview thumbnail :(

**Comment by mertalev** [comment]
There's an option in the Image Settings to use the embedded preview where possible. The result should be better if you enable that, re-run thumbnail generation on all assets and clear browser cache once it's done.

**Comment by MikevE1995** [comment]
This fixed it for me, thanks!

**Comment by jedie** [comment]
> There's an option in the Image Settings to use the embedded preview where possible. The result should be better if you enable that, re-run thumbnail generation on all assets and clear browser cache once it's done.

But this fix only low res thumbnails... What i mean in https://github.com/immich-app/immich/issues/13029#issuecomment-2465799903 above: In detail view of one picture is also only the low res thumbnail used! This will be not fixed, isn't it?

**Comment by ibigbug** [comment]
I'm having the same issue and realized that my failing DNG can't be opened by Windows Photo editor either. though it works fine in Photoshop on Windows

**Comment by louhivuori** [comment]
Confirming that this affects iPhone DNGs from iOS18. The above workaround enabling embedded previews helps until issue is fixed.

Administration > Settings > Image Settings > Prefer embedded preview
and then
Jobs > Generate thumbnails > Missing

**Comment by Riasann** [comment]
Sorry the issue still happend also with the Settings:

"Administration > Settings > Image Settings > Prefer embedded preview
and then
Jobs > Generate thumbnails > Missing"

DNG and HEIC from IOS18 cannot be displayed and no thumbnails.

Server : v1.125.7

But the issue not happend by all IOS18 DNG and HEIC, but seam like happend after i send the images via Whatsapp and then this Copy getting upload via immich IOS App.
So the Modifyed Copy dosent want to load.

The same Image upload from owner IOS18 works.

**Comment by abhijeetviswam** [comment]
I have a slightly different, but possibly related issue.

For DNG's edited with Lightroom, on desktop I don't see any of the edits applied in the thumbnails or full views. On Android mobile also I see the thumbnail doesn't have any edits applied. However, on full view in android mobile shows the edits properly.

As I understand from various reddit threads, RAW files are supposed to show up as, well RAW in immich gallery. So I guess what I'm seeing is probably a bug and not a feature?

But if it works in android, can we get it working everywhere else as well?

https://github.com/user-attachments/assets/00a50804-79db-4e06-b1c0-b597738dc96c

**Comment by carloslockward** [comment]
Just encountered this issue with .dng RAW images taken on an S24 Ultra. Is the team aware of the issue?

**Comment by mertalev** [comment]
There's not much we can do about it since DNG compatibility is related to upstream image processing libraries. The potential solution is to use the Adobe DNG SDK, but that doesn't (officially) support Linux.

**Comment by vazquezjm** [comment]
Same issue with the Samsung S25 Ultra and .DNG images: they don’t even appear in the app’s main feed, nor are they backed up to the server.

**Comment by MK73DS** [comment]
Same issue here with a Samsung Galaxy S24 Ultra's raw dng files. They are backed up but thumbnail generation fails.

**Comment by timbatog** [comment]
Is there any value in providing sample DNG files from my OM-1 Mark II Version 1.2 - both as native RAW and converted in Lightroom DNGs (lossy and lossless)?

I'd love to be able to see my DNGs in Immich but as of today - June 2025 - these just appear as broken thumbnails and break by processing workflow.

**Comment by Kh3nsu** [comment]
Any news on this? I sadly also have some DNG Images from a friend which are just showing up as broken images, but when I download them, they are fine.

**Comment by PseudoResonance** [comment]
Adobe DNG SDK is quite easy to build on Linux, this is not an issue. Libraw already supports Adobe DNG as of snapshot 202403 and can be built with it.

The issue right now is [ImageMagick does not support Adobe DNG](https://github.com/ImageMagick/ImageMagick/issues/8324), so while I have added the DNG SDK to the build image, Immich still wasn't able to unpack my DNGs until ImageMagick support is done.

For reference, [here is my Immich base images repository](https://github.com/PseudoResonance/immich-base-images/tree/dng) I was using to test support.

**Comment by mertalev** [comment]
ImageMagick should only need to interact with libraw and libraw is what should interact with DNG SDK directly, right? Is there a different libraw API it would need to use in this case?

**Comment by PseudoResonance** [comment]
> ImageMagick should only need to interact with libraw and libraw is what should interact with DNG SDK directly, right?

[Nope, it's documented here](https://github.com/LibRaw/LibRaw/blob/ae32c08bfe2374c651176ecf32e07a3751697488/README.DNGSDK.txt#L31). Libraw works with Adobe DNG, but you have to pass the `dng_host` to it yourself, which means that ImageMagick has to build with the SDK too.

[Here's a sample app, dcraw_emu](https://github.com/LibRaw/LibRaw/blob/ae32c08bfe2374c651176ecf32e07a3751697488/samples/dcraw_emu.cpp#L462) that utilizes it.

Also I said it in my ImageMagick issue, but while I could get it done, I'm not sure of ImageMagick's coding standards, so if anyone here knows how they handle C/C++ interop there, it'd be great if it could get implemented. I'm pretty sure it should be relatively simple, but I just spent the past few days learning how all these other SDKs come together and am a little worn out...

Also some additional info for anyone who happens to pass by.

Adobe DNG SDK 1.7.1 is compatible with libjxl up to version 0.8.x, but not 0.9.x. I [made a patch](https://github.com/PseudoResonance/immich-base-images/blob/9a80d18b3e891a0bd89f855a1fe0f1cfc6185c2e/server/sources/libdng-build/dng_jxl.cpp.patch) to build against newer libjxl.

If you want to try building all this yourself, I would advise you to take a look at [some of the fixes I had to do](https://github.com/PseudoResonance/immich-base-images/blob/dng/server/sources/libdng.sh) to get it all building, because Adobe's build scripts are the absolute worst. Hopefully it can serve as a starting point at least to prevent the suffering that is dealing with them.

Edit: It's unlikely that ImageMagick will implement support for the latest libraw until the specification has been finalized in a release, so it's possible that this will not be available for a year.

**Comment by nijahplays** [comment]
Still having this issue on latest with the Samsung S25 Ultra RAW DNG files. They only load if the files are still stored on the uploading device.

**Comment by FlopRe** [comment]
+1 Same issue here on raw pictures from S23 Ultra - but ONLY on a few photos out of hundreds. It's also making mobile app get stuck on backup with it infinitely loading and never finishing.

Edit: Resolved on its own. I quit the app and reopened, and the backup now seems to have completed with no further issues. I also searched for the photo using its filename and can see the thumbnail and full size pic (although it looks terrible converted).

**Comment by wassupluke** [comment]
*+1* same issue here and also unresolved by the aforementioned workaround.

It is severe enough a crash that it brings down my server each time it tries to generate those thumbnails.

The particular images in my case are exposure-bracketed Canon Raw (`.CR2`) files that I've merged in Lightroom Classic which saves them as `.dng` files that are backed up to my NAS and accessed as an external library in Immich, if that helps determine anything. But yes, can confirm that Magick complains the file format is unsupported.

e.g., (hundreds of these)

```bash
[Nest] 7  - 10/12/2025, 9:57:04 PM   ERROR [Microservices:{"id":"d214f0cd-0702-437e-aa6c-2aa299047c77"}] Unable to run job handler (AssetGenerateThumbnails): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/NAS_Lightroom_ro/2023/10 - SENIOR PHOTOS/IMG_9656-HDR.dng' @ error/dng.c/ReadDNGImage/536 (null)
Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/NAS_Lightroom_ro/2023/10 - SENIOR PHOTOS/IMG_9656-HDR.dng' @ error/dng.c/ReadDNGImage/536 (null)
    at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.3/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/usr/src/app/server/dist/repositories/media.repository.js:105:68)
    at descriptor.value (/usr/src/app/server/dist/repositories/telemetry.repository.js:157:39)
    at MediaService.decodeImage (/usr/src/app/server/dist/services/media.service.js:177:59)
    at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:189:44)
    at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:116:25)
```

**Comment by kmilos** [comment]
> I've merged in Lightroom Classic which saves them as .dng files

@wassupluke You could choose to export to an earlier DNG version that is _not_ 1.7 and thus has wider compatibility.

**Comment by QuakeGio83** [comment]
Hi,
I have same problem:

immch: 2025-10-19 05:42:52.428975+00:00�[31m[Nest] 7 - �[39m10/19/2025, 7:42:52 AM �[31m ERROR�[39m �[33m[Microservices:{"id":"b4a7a1dd-c305-4e77-9840-4164acd014c1"}]�[39m �[31mUnable to run job handler (AssetGenerateThumbnails): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file /data/library/admin/2025/2025-10-12/20251012_181310.dng' @ error/dng.c/ReadDNGImage/536 (null)�[39m 2025-10-19 05:42:52.429081+00:00Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file /data/library/admin/2025/2025-10-12/20251012_181310.dng' @ error/dng.c/ReadDNGImage/536 (null)


Are you working to fix it? I'm using Samsung S25 Ultra and to keep best pictures quality I use RAW file in expert mode when take pictures

**Comment by quangtdev** [comment]
same issue

**Comment by danieLayosh** [comment]
I have the same error when taking a photo on iphone 17 pro max raw, dng files are backup and can be viewed in the mobile app, but when opening on the web those dng photos cant be processed and it says image "loading error"
I started looking around for some clues about what can cause it. I thought maybe the format is not supported on the web version or maybe web browsers can't even open DNG files but no luck for me.

**Comment by traktuner** [comment]
> I have the same error when taking a photo on iphone 17 pro max raw, dng files are backup and can be viewed in the mobile app, but when opening on the web those dng photos cant be processed and it says image "loading error" I started looking around for some clues about what can cause it. I thought maybe the format is not supported on the web version or maybe web browsers can't even open DNG files but no luck for me.

Same issue for me, same phone.

This is the error I get in the logs, I guess the new cameras are not yet supported by libraw (@mertalev) ? 

[Nest] 6  - 11/11/2025, 6:23:27 AM   ERROR [Microservices:{"id":"XXXXXXXX-XX-XX-XX"}] Unable to run job handler (AssetGenerateThumbnails): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/usr/src/app/upload/library/XXXX/XXXX/XXXX-XX-XX/IMG_XXXX.dng' @ error/dng.c/ReadDNGImage/536 (null)
Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/usr/src/app/upload/library/XXX/XXX/XXXX-XX-XX/IMG_XXXX.dng' @ error/dng.c/ReadDNGImage/536 (null)
    at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.4/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/usr/src/app/server/dist/repositories/media.repository.js:105:68)
    at MediaService.decodeImage (/usr/src/app/server/dist/services/media.service.js:177:59)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:190:44)
    at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:116:25)
    at async JobService.onJobRun (/usr/src/app/server/dist/services/job.service.js:199:30)
    at async EventRepository.onEvent (/usr/src/app/server/dist/repositories/event.repository.js:91:13)
    at async /usr/src/app/server/node_modules/.pnpm/bullmq@5.61.2/node_modules/bullmq/dist/cjs/classes/worker.js:528:32

**Comment by PseudoResonance** [comment]
TLDR: Do you know people who work at Adobe? Please help us get in contact with their lawyers. Thank you.

Please, for the sake of everyone's notifications, stop commenting what we already know. Several people here have already very clearly outlined the exact issue, I gave links to the projects we are waiting on, as well as an estimated time frame (1 year).

Sending everyone here a million emails for every comment is not going to speed up the process.

In fact, the unfortunate reality is that these filles may never be decoded in Immich due to Adobe's communication failures and lack of any clear licensing.

If you really want to help make it happen, please put your valuable time towards contacting Adobe and figuring out the license for their DNG SDK. Until that happens, Immich can't decode these files.

[I put some information about it here](https://github.com/ImageMagick/ImageMagick/issues/8324#issuecomment-3226825601). I spent hours talking to people and getting nowhere...

I apologize for the harsh message, but I just want to get the word out that we need to be speaking with people at Adobe. Your efforts are wasted here.

**Comment by wassupluke** [comment]
> There's not much we can do about it since DNG compatibility is related to upstream image processing libraries. The potential solution is to use the Adobe DNG SDK, but that doesn't (officially) support Linux.

Understood (also understood is the most recent comment above me). In the short term, can we get an option to ignore .dng/.DNG file extensions so I can load my external library without crippling my immich server while it fails to process hundreds of dngs?

**Comment by PseudoResonance** [comment]
> Understood (also understood is the most recent comment above me). In the short term, can we get an option to ignore .dng/.DNG file extensions so I can load my external library without crippling my immich server while it fails to process hundreds of dngs?

Unfortunately, no. Because it actually can read the older DNG file formats, so if it skipped all .dngs, it would miss potentially valid files.

Maybe an option could be added? I don't know, I'm not a maintainer. I just have a personal interest in solving this.

Edit: Just to clarify one thing, Adobe DNG SDK Linux support is a non issue. It's trivial to build and works as expected. The only thing truly stopping this is Adobe's licensing.

**Comment by AGJSpiegelenberg** [comment]
Isn't this the DNG SDK license agreement ? https://www.adobe.com/support/downloads/dng/dng_sdk_eula_mac.html

**Comment by PseudoResonance** [comment]
> Isn't this the DNG SDK license agreement ? https://www.adobe.com/support/downloads/dng/dng_sdk_eula_mac.html

Maybe, but that's not good enough in law. That page's download link no longer functions and the last snapshot on Archive.org from 2020 is showing 1.5.1 available at the time. 1.5.1 was, I believe, the last download to include that license in the files. We need the licensing terms for 1.7.1, released in 2023.

**Comment by ptr727** [comment]
None of my DNG's created by Lightroom as part of RAW import workflow display, nor does any of my iPhone DNG images?

**Comment by timbatog** [comment]
To add to the complexity of understanding DNGs (sorry!) I've gone for DNG v16 and Lossy compression for many of those 'don't need forever' DNGs (coming from OM-1 mk2 ORF original RAW files) . Needless to say they don't render in Immich, which is a shame.

Who can we poke at adobe to get the licensing terms that we need? Do they have a team? I'll gladly chase....

**Comment by PseudoResonance** [comment]
> To add to the complexity of understanding DNGs (sorry!) I've gone for DNG v16 and Lossy compression for many of those 'don't need forever' DNGs (coming from OM-1 mk2 ORF original RAW files) . Needless to say they don't render in Immich, which is a shame.
> 
> Who can we poke at adobe to get the licensing terms that we need? Do they have a team? I'll gladly chase....

Feel free to try... I emailed a bunch of legal-related addresses, called several support agents and asked around if anyone I knew had contacts, but still wasn't able to get in touch.... Presumably only the legal department could help though.

**Comment by CrazyWolf13** [comment]
Hi,
I've been following this thread since some time, as I have the same issue with raw images taken from my s25 ultra.

This was not only on immich, also on other Opensource Photo Editing Software, but today I noticed something strange in immich, it seemed like most of my .dng shots have now a working preview, although it's nearly compledtely dark, you can see the actual image, so I guess for raw shots from the samsung s25 ultra taken via Expert Raw App seem to be working, although it takes a long time to render them.

What I noticed just some days ago, that Proton Drive, which is also opensource seems to also render the S25 Ultra raw shots fine now, earlier this was not working, but since their new update they render, so maybe it's possible to look at how proton drive is rendering them? (Linking their sourcecode: https://github.com/ProtonDriveApps/android-drive)

**Comment by noah-buehler** [comment]
I have a similar issue here, RAW files shot on the S23+ (Expert RAW) didn't render at all until I enabled full size image geneartion. Sadly however even when I did the DNG files look really dark and greensish... Toggeling "Prefer embeded preview doesn't fix it. (I assume those raw files don't have a embeded preview?) It works well with my Fuji RAW files... [Here](https://immich.noahbuehler.com/share/drnfRbSz9UWqWz-D28wy7Zwg3vAXJkEnRbFw2DXyg5V4cEQm6F1-vdOBiaQg7NWGGxo) are some sample Photos hosted on immich so you see how the issue looks and you can download them if it helps.

**Comment by PseudoResonance** [comment]
> What I noticed just some days ago, that Proton Drive, which is also opensource seems to also render the S25 Ultra raw shots fine now, earlier this was not working, but since their new update they render, so maybe it's possible to look at how proton drive is rendering them? (Linking their sourcecode: https://github.com/ProtonDriveApps/android-drive)

I glanced through it and don't see anything particularly interesting. I've never used the app, so I don't really know where to look, but [this area appears to be responsible for drawing the images](https://github.com/ProtonDriveApps/android-drive/blob/d3e42bb210ae5aaf15ed54b9ced16554f1aae814/photos/presentation/src/main/kotlin/me/proton/android/drive/photos/presentation/component/MediaItem.kt#L115)? It's just using Android APIs though, which doesn't help on iOS or browsers.

**Comment by psychedelicu** [comment]
2026 and still have this issue..

**Comment by mrkbaker** [comment]
Would love this to be fixed as many of my photos are .dng photos. I find alot of my library just showing "Error loading image"

**Comment by Oshyan** [comment]
I'm also running into this on a metric ton of my older images (30k or more, poor Pentax/Ricoh). Aside from just getting a fix, I'm unclear as to whether this is a fundamental tech limitation or just as much a licensing/legality one (akin to the old video codec rights issues that were a bigger deal some years back across many OS tools). In particular I want to understand whether there is currently a workaround possible "at your own risk" (legally/license-wise) by somehow including or referencing the Adobe SDK in my own private/local Immich install. I got that impression somewhere in this or a related thread, but I'm not at all sure it's correct (and if it is, how to do it).

**Comment by PseudoResonance** [comment]
> Aside from just getting a fix, I'm unclear as to whether this is a fundamental tech limitation or just as much a licensing/legality one (akin to the old video codec rights issues that were a bigger deal some years back across many OS tools).

Immich is waiting on ImageMagick. [ImageMagick is (correctly) being hesitant](https://github.com/ImageMagick/ImageMagick/issues/8324#issuecomment-3226776907) to add support, due to licensing issues.

LibRaw has released the feature now, however it still requires ImageMagick to include AdobeDNG code themselves, and feed it back into LibRaw for processing, so even if LibRaw is okay with including AdobeDNG SDK stuff regardless of the licensing, ImageMagick also has to for it to work.

As for what you can do now, at the moment, you can build LibRaw with the Adobe SDK, however ImageMagick has no support for it, so you would have to add it yourself and maybe open a PR to ImageMagick. Then Immich's Docker image build scripts would need to be updated to build the Adobe SDK, LibRaw, and ImageMagick with the patches.

**Comment by mertalev** [comment]
It's ultimately a licensing question regardless of ImageMagick. If it were clear that it's okay to use the SDK, then we could patch imagemagick/libraw to use it since the code change seems simple. But if that were established, I assume ImageMagick would also be fine with supporting it.

**Comment by Oshyan** [comment]
Thanks for the clarification! It turns out I know someone who works at Adobe as a developer. I haven't asked him about this yet, and I have no idea if he can contact the right people (seems like the legal dept. is what's needed). But if I were to ask him, can you tell me _exactly_ what is needed? Is it a current version of the license agreement that explicitly covers a usage like ImageMagick/Immich?

**Comment by mertalev** [comment]
> Thanks for the clarification! It turns out I know someone who works at Adobe as a developer. I haven't asked him about this yet, and I have no idea if he can contact the right people (seems like the legal dept. is what's needed). But if I were to ask him, can you tell me _exactly_ what is needed? Is it a current version of the license agreement that explicitly covers a usage like ImageMagick/Immich? 

The SDK stopped including a license file after 1.5.1. The current version of the SDK needs to include this license, or the download page has to at least say the SDK is licensed under X.

**Comment by Oshyan** [comment]
Yeah, I looked into all of this again myself just now, including checking the SDK code itself, which still references a non-existent license file that's supposed to accompany the SDK. Very odd that they would continue to provide the SDK but not include a license file or even reference one for the SDK on the download page! At least the old download page had a license agreement gate, the current one doesn't even have that. 

My friend said he'd ask someone at Adobe if he can. In the meantime my workaround _may_ be to regenerate higher quality embedded thumbnails in all my old DNGs and turn on "prefer embedded image".

**Comment by RobAtAdobe** [comment]
@Oshyan My name is Rob and I work at Adobe. I've confirmed that our website currently has an issue. I've relayed it to the right team internally who look at addressing this. Thank you for relaying and my apologies for the inconvenience.

**Comment by ptr727** [comment]
Fyi per a contact at Adobe (maybe we now have multiple people asking Adobe to fix the license embedding ;) ):

[DNG SDK License Agreement](https://www.adobe.com/support/downloads/dng/dng_sdk_eula_win.html?msockid=1c40437b975c6acc1a2255c296126b06)

[DNG Specification patent license](https://helpx.adobe.com/camera-raw/digital-negative.html#dng)

**Comment by PseudoResonance** [comment]
> Fyi per a contact at Adobe (maybe we now have multiple people asking Adobe to fix the license embedding ;) ):
> 
> [DNG SDK License Agreement](https://www.adobe.com/support/downloads/dng/dng_sdk_eula_win.html?msockid=1c40437b975c6acc1a2255c296126b06)
> 
> [DNG Specification patent license](https://helpx.adobe.com/camera-raw/digital-negative.html#dng)

First link is invalid as it's for an older version.

Second link is for the specification, not the SDK. That's unfortunately the same stuff that Adobe pointed me to.

**Comment by ptr727** [comment]
I was pointed to [this updated page](https://helpx.adobe.com/camera-raw/digital-negative.html), but the SDK zip still does not include a license file.

**Comment by Oshyan** [comment]
> I was pointed to [this updated page](https://helpx.adobe.com/camera-raw/digital-negative.html), but the SDK zip still does not include a license file.

That page says it was updated January 30th which is well before much of the recent discussion here. Was it implied by whomever pointed you there that it was more recently updated than that?

Hopefully Rob got the ball rolling internally at Adobe, but it doesn't seem like anything has changed _yet_. And I'm not surprised, if it involves legal to approve the license they embed with the download it could take a while, and even if it doesn't need legal sign-off (existing, already-approved license text), it could take time. I'll check back in a week. 😄

**Comment by ptr727** [comment]
I was told that this thread is being looked at by Adobe, and it was suggested that the ask from Immich to Adobe should be clearly spelled out, e.g. "please include the actual DNG license file in the DNG SDK zip file"

**Comment by mertalev** [comment]
> I was told that this thread is being looked at by Adobe, and it was suggested that the ask from Immich to Adobe should be clearly spelled out, e.g. "please include the actual DNG license file in the DNG SDK zip file"

Thanks for the update! The ask is exactly what you mention: it should include a LICENSE file in the DNG SDK zip. That would be enough to resolve the current licensing ambiguity for Immich and other projects.

**Comment by ptr727** [comment]
> Thanks for the update! The ask is exactly what you mention: it should include a LICENSE file in the DNG SDK zip. That would be enough to resolve the current licensing ambiguity for Immich and other projects.

I was just told that a new SDK with license file included will be published around the 10th.

**Comment by AGJSpiegelenberg** [comment]
It seems the current SDK download includes a LICENSE.txt
[Download the Adobe DNG SDK](https://www.adobe.com/go/dng_sdk) (1.7.1  Build 2471, January 29, 2026)

**Comment by Oshyan** [comment]
> It seems the current SDK download includes a LICENSE.txt [Download the Adobe DNG SDK](https://www.adobe.com/go/dng_sdk) (1.7.1 Build 2471, January 29, 2026)

That's progress! Unfortunately I think the license text may be incompatible with the AGPL-3.0 license of Immich, as well as ImageMagick's license. I don't know if it at least gives us workarounds, hopefully so. At the least it addresses the odd lack of license for the SDK and makes clear its terms of use.

**Comment by PseudoResonance** [comment]
> That's progress! Unfortunately I think the license text may be incompatible with the AGPL-3.0 license of Immich, as well as ImageMagick's license. I don't know if it at least gives us workarounds, hopefully so. At the least it addresses the odd lack of license for the SDK and makes clear its terms of use.

From my terribly rough understanding, GPL 3.0 doesn't necessarily exclude [adding a library that isn't GPL compatible](https://www.gnu.org/licenses/gpl-faq.en.html#FSWithNFLibs). Obviously this is just their guidelines, not the license, but they seem to encourage you to find alternatives, however it isn't explicitly banned.

**Comment by bo0tzz** [comment]
In the Immich dev team we expect this is probably fine, but we've asked someone with actual legal knowledge to take a look at this just in case, we'll report here when we hear back.

**Comment by hegemon2** [comment]
Following. Prefer embeded preview fixed this for me (iPhone/iOS RAWs) because otherwise I was getting broken images.

 I actually think I unintentionally have deleted _many_ "broken" files from Immich believing they somehow got broken in the upload, not realizing that I was unwittingly deleting my RAWs from immich. So that's annoying.  

What's the long-term fix for this? Obviously it would be preferable (I think?) to be able to at least view a 100% quality JPG in browser. And I suppose this raises a related question: how are edits made on the device to the RAW handled by Immich?

**Comment by PseudoResonance** [comment]
I just wanted to thank everyone for working on getting in touch with Adobe and finally moving this issue forward! Also of course the people at Adobe who worked on making this happen.

**Comment by magiceupho** [comment]
Hi,
I've a similar problem with compressed and uncompressed DNG files from PhotoLab 9.
```
STDERR
[Nest] 7  - 03/24/2026, 2:55:09 PM   ERROR [Microservices:{"id":"e346fb6e-9c83-47da-b813-ac0adac191bd"}] Unable to run job handler (AssetGenerateThumbnails): Error: Input buffer contains unsupported image format
STDERR
Error: Input buffer contains unsupported image format
STDERR
at Sharp.metadata (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/input.js:637:17)
STDERR
at MediaRepository.getImageMetadata (/usr/src/app/server/dist/repositories/media.repository.js:264:95)
STDERR
at MediaService.shouldUseExtractedImage (/usr/src/app/server/dist/services/media.service.js:599:62)
STDERR
at MediaService.extractImage (/usr/src/app/server/dist/services/media.service.js:172:39)
STDERR
at async MediaService.extractOriginalImage (/usr/src/app/server/dist/services/media.service.js:191:45)
STDERR
at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:213:32)
STDERR
at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:153:25)
STDERR
at async JobService.onJobRun (/usr/src/app/server/dist/services/job.service.js:51:30)
STDERR
at async EventRepository.onEvent (/usr/src/app/server/dist/repositories/event.repository.js:91:13)
STDERR
at async /usr/src/app/server/node_modules/.pnpm/bullmq@5.70.4/node_modules/bullmq/dist/cjs/classes/worker.js:570:32
```
My server is on 2.6.1 version with Docker.
I've turned the Prefer embeded preview option on, but it doesn't help.

Is there a solution?

Thank you

**Comment by mertalev** [comment]
Unfortunately, we got legal feedback that the SDK license adds restrictions that make it incompatible with AGPLv3. In particular, it means users agree to "defend, indemnify and hold harmless Adobe against any losses, damages and costs arising from the claims, lawsuits or other legal actions of such distribution" and restricts the governing law and jurisdiction to California.

The options are to either not use the SDK, somehow isolate it in a way that doesn't make usage constitute a combined work, or for the terms to change. Isolating it is likely to add substantial complexity to the image processing stack, so I'm not sure what the path forward is for this.

**Comment by PseudoResonance** [comment]
Ah, that's unfortunate, but thanks for the update!

It does say that it only affects "commercial products," however Immich takes donations. Is that a "commercial product"?

Regardless, it sounds like Immich probably cannot package it into the builds themselves, and either way, ImageMagick has always been the real blocker, not Immich. It probably conflicts with their licensing as well...

Perhaps the only way would be to somehow avoid "Any download, installation, use, reproduction, modification or distribution of the Software or Documentation" except for the end user. ImageMagick would still need the library headers to build against though, and I wonder if headers count as "software or documentation" or not... I just wish companies like Adobe would release stuff with more permissive licenses...

**Comment by webysther** [comment]
> Unfortunately, we got legal feedback that the SDK license adds restrictions that make it incompatible with AGPLv3. In particular, it means users agree to "defend, indemnify and hold harmless Adobe against any losses, damages and costs arising from the claims, lawsuits or other legal actions of such distribution" and restricts the governing law and jurisdiction to California.
> 
> The options are to either not use the SDK, somehow isolate it in a way that doesn't make usage constitute a combined work, or for the terms to change. Isolating it is likely to add substantial complexity to the image processing stack, so I'm not sure what the path forward is for this.

Isolate is actually possible (and easy) using the same method of LinuxServer did, using mods:

https://docs.linuxserver.io/images/docker-code-server/#docker-mods

Just add a feature flag to use external dns sdk and give the user the way to add this using mods, this will make decoupled from the project.

**Comment by PseudoResonance** [comment]
Primarily, ImageMagick cannot currently use the SDK because it's C++ only and has no C headers. Most likely, someone would need to build and maintain (and submit to ImageMagick?) C compatibility patches for the SDK to allow it to be used. This is still assuming that the SDK license does not conflict with ImageMagick's license and that they will find a legally acceptable way to provide support.

For Immich, someone needs to finish and maintain SDK build scripts, [such as these](https://github.com/PseudoResonance/immich-base-images/blob/dng/server/sources/libdng.sh), to actually compile it for Linux. Since Immich does not want to distribute it, there would need to be a mechanism of some sort to dynamically load it with a feature flag, or a mechanism by which users can inject it themselves as mentioned by webysther.

Also, most likely ImageMagick would also want to utilize such a build script to be able to test/build against the SDK in the first place. It's unfortunately a pain to get building out of the box, and Adobe does not support Linux at all.

There's still quite a few hurdles before this could ever be seen in Immich...

**Comment by ItzDerock** [comment]
instead of trying to dynamically load it into the main immich server, could immich implement some sort of routing/microservices system where we define either via env or dashboard settings that all DNGs should be processed by a microservice available at some passed url (like `http://immich-dng-handler`). That server can be distributed separately, and could even be a separate project not maintained by the immich team. All that's required is immich have some api spec that the external handlers would need to follow (i.e. a post endpoint /thumbnail/generate). And maybe instead of only having this for dng's, it's something where you specify a mime type + url, so if there's ever more formats that are under proprietary licenses, we can have a community maintained "extension" list.

so for a user who wants dng support, they'd add a separate microservice to their compose stack, and register it with immich.

I'm not familiar with immich's working internals, so not sure how easy or hard this would be to implement, just my quick thoughts on the matter. but just from an oop perspective, something akin to an abstract class ImageHandler that has impls ImageMagickHandler vs ExternalImageHandler, and a basic router that chooses depending on mime type.

**Comment by mertalev** [comment]
This is just a shower thought, but we've been working on a plugin system where an installed WebAssembly plugin can hook into certain events (like after an asset is uploaded, etc.). It could maybe be possible to extend that so it can be a delegate for image processing. No promises on whether this will happen, but it could be a way forward.

**Comment by PseudoResonance** [comment]
Both would definitely be good in my opinion. Of course too, if we're not sending all images to it and only DNGs, we could use a more specialized tool for it. Would still have limitations though I guess...

Perhaps the easiest would be a way for a "plugin" to "transcode" files for thumbnails and extra processing and stuff? Leave the original DNG (or other obscure file format) intact, but Immich can otherwise work with the transcoded file?

Could be expanded to other weird formats that ImageMagick doesn't/can't support too, if there even are any! I'm imagining it would probably still be a huge undertaking though... I'd love to see DNGs working in Immich, but I'm still very happy with it regardless!

**Comment by Oshyan** [comment]
Unfortunately this is fundamentally a blocker for me using Immich in any real way as a have a huge catalog of DNG-only images from various generations of cameras, several of which aren't supported without Adobe's SDK. It certainly seems ideal if the need for this kind of support can be met by a more generalized system/service/aspect of Immich that can also address other possible current and future needs as well! That makes it a much more attractive dev target. So I'm hopeful for the plugin system being able to accommodate this!

**Comment by Dowan35** [comment]
Its been 2 years that we have this issue, any workaround guys ? I have the problem using Expert Raw app on S25 phone, thanks

**Comment by PseudoResonance** [comment]
@mertalev I made a [hacky example here](https://github.com/PseudoResonance/immich-base-images/commit/386aaa833aa5e7778bb4d27c7d7d3d932bc172ba) that just bypasses the mess with ImageMagick and/or libvips. Sorry for the wall of text, just trying to come up with something.

I will not be making a PR, because this isn't suitable in its current state, however, I wanted a proof of concept sort of, and I was wondering if the Immich team has any ideas on a way that something similar can be done that allows the use of the DNG SDK for users that want it? The commit is just for ideas on what kind of work is necessary to build the tools.

It mainly boils down to a simple patch for libraw, but there's quite a dependency chain, so it seems like it'd be difficult to find a way for Immich to not include the DNG SDK, but allow users to inject it on their side or something. Whatever satisfies the legal constraints. Perhaps the ultimate answer will be that community members just need to modify the Immich images and inject the final libvips and dependencies for their own use?

Edit: [Here's a really simple example](https://github.com/PseudoResonance/immich-dng-images) workflow for customizing the images with the DNG support. Immich was able to thumbnail my phone's DNGs properly. (I previously tested just libvips thumbnailing a few sample DNGs I downloaded from comments here as well.)  Although, do not use this unless you want to test stuff and don't mind things breaking.

### Why does the commit look like this
Adobe DNG SDK is a bit of a pain to build on Linux as Adobe doesn't provide proper build scripts. Some hacking is required to modify the included scripts and build it more easily. Additionally, the DNG SDK requires libxmp and is part of the build process, so I included it.

Libraw then needs to be built using the DNG SDK, so some customization is needed to build that, as well as the other dependencies that DNG files can rely on, like JXL.

The main issue though is that libraw only allows the use of the DNG SDK through the C++ API, but libvips and ImageMagick are primarily C. We don't particularly need the  customization features though, so I opted to just patch libraw to create its own DNG SDK instance with default options. This allows any consumer library to just work without the expected requirements.

Additionally, [this PR to replace ImageMagick](https://github.com/immich-app/base-images/pull/312) made me realize that libvips can talk to libraw directly now, so having libraw just work out of the box on either is more convenient for this hacky solution.

**Comment by mertalev** [comment]
@PseudoResonance I actually packaged the DNG SDK into [libdngsdk](https://github.com/mertalev/libdngsdk) for Linux some time ago in anticipation of being able to use it. My thought was to patch/upstream libvips to be built with it like its other pkg-config libraries and have libvips pass the SDK handle to libraw. This is not official advice (and probably bad advice in general), but it's possible to build libvips/imagemagick yourself consistent with how the [base image](https://github.com/immich-app/base-images.git) builds it and mount your .so to replace the one in the container.

**Comment by mertalev** [comment]
The libvips branch I tested libdngsdk with is [here](https://github.com/mertalev/libvips/commit/288ff242b9fb6d4397e4e6ba907015479651ae5a) if that helps.

**Comment by PseudoResonance** [comment]
> but it's possible to build libvips/imagemagick yourself consistent with how the [base image](https://github.com/immich-app/base-images.git) builds it and mount your .so to replace the one in the container.

That was my thought on what might be the best way forward. Exactly what's [going on here](https://github.com/PseudoResonance/immich-dng-images/blob/main/build.sh), however by hacking libraw insead of libvips.

Only problem with directly mounting at runtime though is the need to run ldconfig prior to Immich booting.

Edit: One thing I'd appreciate from Immich though is to generate the fullsize images along with the thumbnails for image formats that the browser can't display natively, like DNG. Currently, the DNG is thumbnailed, but I can only view the smaller thumbnail image on Immich, since the original one isn't available to view. Otherwise, metadata, thumbnails, etc all work perfectly.

**Comment by mertalev** [comment]
> Only problem with directly mounting at runtime though is the need to run ldconfig prior to Immich booting.

I haven't tested, but I don't think you need `ldconfig` if it's installed to `/usr/lib`.

> One thing I'd appreciate from Immich though is to generate the fullsize images along with the thumbnails for image formats that the browser can't display natively, like DNG. Currently, the DNG is thumbnailed, but I can only view the smaller thumbnail image on Immich, since the original one isn't available to view. Otherwise, metadata, thumbnails, etc all work perfectly.

If you enable full-size images in the Image Settings, it does exactly this.

**Comment by PseudoResonance** [comment]
> If you enable full-size images in the Image Settings, it does exactly this.

I completely missed that, thank you! Works quite well, however it appears there isn't a way to view originals but fallback to the fullsize preview? Or something like the HTML `picture` element to automatically fallback if the browser believes a MIME type is unsupported.

> I haven't tested, but I don't think you need ldconfig if it's installed to /usr/lib.

Maybe. I installed to `/usr/local/lib` since that's where the originals were. It complained about not being able to find libraw_r.so.25 though.
Edit: Quick test seems to suggest it works.

**Comment by mertalev** [comment]
> I completely missed that, thank you! Works quite well, however it appears there isn't a way to view originals but fallback to the fullsize preview? Or something like the HTML picture element to automatically fallback if the browser believes a MIME type is unsupported.

There's a whitelist of formats considered web-compatible and it will load the original instead of the full-size image if the original is in one of these formats. The code also tests for JPEG XL and HEIF support and allows originals in those formats if they work.

**Comment by PseudoResonance** [comment]
> There's a whitelist of formats considered web-compatible

Ah yeah I guess that's why the web-compatible DNGs refuse to show originals now. Probably not much that could be done about that. Thanks for the information though!

**Comment by PseudoResonance** [comment]
With the workflows coming, perhaps some kind of a workflow to write/overwrite both thumbnails and full-size images via some kind of web API could be helpful? Then you could run a basic sidecar container with the DNG SDK and keep it entirely separate instead of having to modify the Immich container somehow.

A workflow that's called on any image thumbnail/fullsize generation? Or just use on asset upload and have a step that can set the images? I haven't looked at how Immich works internally and kind of doubt either of these would be feasible, but was just an idea.

**Comment by Sieboldianus** [comment]
I appear to have this problem with v2.7.5 and Nikon `nef` files (D800):
```
immich_server            | [Nest] 7  - 06/29/2026, 6:08:00 AM   ERROR [Microservices:{"id":"01131012-525a-40c7-9077-1715c1306cf5"}] Unable to run job
handler (AssetGenerateThumbnails): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/media/alex_
photos/01_RAW/20110313_030/wrong wb/_DSC3267_0.nef' @ error/dng.c/ReadDNGImage/508 (null)
```

**Comment by F1R3F1GHTR** [comment]
> I appear to have this problem with v2.7.5 and Nikon `nef` files (D800):
> 
> ```
> immich_server            | [Nest] 7  - 06/29/2026, 6:08:00 AM   ERROR [Microservices:{"id":"01131012-525a-40c7-9077-1715c1306cf5"}] Unable to run job
> handler (AssetGenerateThumbnails): Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/mnt/media/alex_
> photos/01_RAW/20110313_030/wrong wb/_DSC3267_0.nef' @ error/dng.c/ReadDNGImage/508 (null)
> ```

Same here. I have uploaded my latest Z5 NEF Files to Immich and they are not showing up. With old Files Uploaded before back in 2025 I still have the thumbnails.


_[Nest] 258  - 07/06/2026, 11:27:52 AM   ERROR [Microservices:{"id":"f98b7bc9-24cb-4b5b-ba76-dd05ead2ae37"}] Unable to run job handler (AssetGenerateThumbnails): Error: Input file has corrupt header: dcrawload: unable to read: Unsupported file format or not RAW file
Error: Input file has corrupt header: dcrawload: unable to read: Unsupported file format or not RAW file
    at Sharp.toBuffer (/app/immich/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/app/immich/server/dist/repositories/media.repository.js:110:68)
    at MediaService.decodeImage (/app/immich/server/dist/services/media.service.js:185:59)
    at async MediaService.extractOriginalImage (/app/immich/server/dist/services/media.service.js:196:44)
    at async MediaService.generateImageThumbnails (/app/immich/server/dist/services/media.service.js:212:32)
    at async MediaService.handleGenerateThumbnails (/app/immich/server/dist/services/media.service.js:152:25)
    at async JobService.onJobRun (/app/immich/server/dist/services/job.service.js:78:30)
    at async EventRepository.onEvent (/app/immich/server/dist/repositories/event.repository.js:91:13)
    at async /app/immich/server/node_modules/.pnpm/bullmq@5.78.1/node_modules/bullmq/dist/cjs/classes/worker.js:589:32_

**Comment by Oshyan** [comment]
This is the runtime-mount variant @PseudoResonance suggested back in June ("mount your .so to replace the one in the container... by hacking libraw instead of libvips"), taken the rest of the way and packaged so anyone can run it. Their libraw host-creation patch is what makes it possible at all: because the consumer (libvips/sharp) needs zero changes and the DNG SDK + XMP toolkit link statically, the entire fix is two patched `libraw` .so files.

One good surprise along the way: the ldconfig concern raised then doesn't actually have any impact. `LD_LIBRARY_PATH` takes precedence over the ldconfig cache, so the patched `libraw_r` shadows the stock one with no ldconfig run and no image modification at all. This means we can keep the stock image, we have just one compose override file, and it's immediately reversible.

I've packaged that as a single self-contained script and published on Github: **[immich-dng-extended](https://github.com/Oshyan/immich-dng-extended)**. Drop it next to your `docker-compose.yml` and run `./immich-dng-extended.sh apply`. It:

- detects your exact Immich version from the running container and fetches the matching `base-images` build recipes, so the build is ABI-identical to your image
- prompts you to accept Adobe's DNG SDK license, downloads the SDK from Adobe's server to your machine (nothing Adobe-owned is redistributed, which is the licensing constraint that keeps this out of Immich proper), then compiles the patched libraw locally inside Docker
- activates it with a `docker-compose.override.yml` that mounts the libs read-only and prepends one path to `LD_LIBRARY_PATH`; the stock image is not modified
- health-checks itself after activation (API up, patched lib actually loaded, a real image pushed through the sharp/libvips pipeline) and automatically rolls back to stock if anything fails
- `./immich-dng-extended.sh remove` is an immediate return to stock; after an Immich update you just re-run `apply`, which rebuilds only if the server image actually changed

So far it's verified on exactly one setup, mine (v3.0.1, arm64, Docker on macOS): Lightroom HDR and pano-stitch DNGs that stock Immich rejects ("Unsupported file format or not RAW file") thumbnail correctly, and a control ARW renders byte-identically before/after (PROOF.md in the repo has the full write-up). It seems to be working well, but one server and one architecture is not much of a sample. If you're comfortable testing, I'd really appreciate people trying it on other setups (amd64 especially) and reporting back. Feedback and PRs are very welcome on the repo. The risk should be low: the build phase can't touch your running server, the script rolls itself back to stock automatically if its post-activation checks fail, and `./immich-dng-extended.sh remove` (or just deleting the override file) returns you to stock at any time.

@PseudoResonance, two small correctness notes on the patch itself in case you iterate on it: the auto-created `dng_host` leaks (nothing frees it, and the destructor can't tell an auto-created host from an externally-set one; an `owns_dnghost` flag cleared in `set_dng_host()` would fix it), and the three duplicated `if (!dnghost) create_dng_host();` insertions could collapse into one lazy accessor.

To the maintainers: no ask here except, someday, a blessed overlay dir + entrypoint hook in the image, which would turn tools like this from "override + env var" into "drop files in a folder." Everything else works with the images exactly as shipped.

**Comment by PseudoResonance** [comment]
> [@PseudoResonance](https://github.com/PseudoResonance), two small correctness notes on the patch itself in case you iterate on it: the auto-created `dng_host` leaks (nothing frees it, and the destructor can't tell an auto-created host from an externally-set one; an `owns_dnghost` flag cleared in `set_dng_host()` would fix it), and the three duplicated `if (!dnghost) create_dng_host();` insertions could collapse into one lazy accessor.

Ah yeah... I figured it might and forgot to look at it again, because I was just trying to see if it worked at the time. I'll look again sometime.

> To the maintainers: no ask here except, someday, a blessed overlay dir + entrypoint hook in the image, which would turn tools like this from "override + env var" into "drop files in a folder." Everything else works with the images exactly as shipped.

Didn't think of that myself, but this is probably the best solution imo. If Immich just copies everything from `/somepath` to `/`, on startup or something, you can use an init container to automatically generate the built libraries, store them in a path for Immich, and then Immich will boot after and copy those libraries in before starting. A solution like this would be needed for Kubernetes too, since dealing with the files directly via some mount is an anti-pattern in Kubernetes.

Edit: Of course, another option is to simply subpath mount in Kubernetes. I believe Docker can do this too, but don't remember for sure. Then you can generate the libs in an init container, and mount them into the Immich container from some shared location between the two containers.

**Comment by jstehlik** [comment]
@Oshyan Thanks for progressing on the solution, I can confirm that this way works for me in my homelab kubernetes stack on amd64. Although I had used Codex to vibecode me an init-container image, which builds the library the same way, it does work then mounting the compiled libs and adding a precedent path to LD_LIBRARY_PATH.

Only thing I've noticed is, that my DNGs from Lightroom that I created by using the HDR stacking from my bracketed photos (was on a trip in december and wanted to pull as much details based on lighting conditions) are a bit dark, but it seems that's because how immich prioritizes embedded previews. But the important thing is there's no more "error loading image" :-)

@PseudoResonance I am not sure if subpath is better, since that would mean a specific filename would always have to be specified to be replaced no? But for the purposes of learning/running in homelab, doing a volumeMount works for me fine.

So for my specific case, in case it helps someone, something like this patch works for me in my deployment:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: immich-server
  namespace: immich
spec:
  template:
    spec:
      initContainers:
        - name: install-dng-libraw
          image: <my homelab registry>/immich-dng-libraw:v3.0.1 #simple compilation image based on Oshyan's script
          imagePullPolicy: IfNotPresent
          args:
            - /overlay/lib
          resources:
            requests:
              cpu: 100m
              memory: 128Mi
          securityContext:
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: true
          volumeMounts:
            - name: dng-libraw-overlay
              mountPath: /overlay/lib
      containers:
        - name: immich-server
          env:
            - name: LD_LIBRARY_PATH
              value: /opt/dng/lib:/usr/lib/jellyfin-ffmpeg/lib:/usr/lib/wsl/lib
          volumeMounts:
            - name: dng-libraw-overlay
              mountPath: /opt/dng/lib
              readOnly: true
      volumes:
        - name: dng-libraw-overlay
          emptyDir: {}
```

**Comment by mertalev** [comment]
Hmm, why would you need an overlay and init container? Couldn't you just extend the server image and make any changes you want?

**Comment by Merlin-2504** [comment]
Since Immich version **3.0.2**, errors have occurred when loading certain DNG files. The issue affects only the web interface and can be reproduced across different browsers—including Chrome and Safari on both desktop computers and iPhones. However, the same images display without issues in the Immich app on the iPhone.
The problem is limited to **unedited DNG files** captured directly with an **iPhone 17 Pro**. The images have not been edited or processed in Lightroom or any other software.
The **"Prefer embedded preview"** option is already enabled in the image settings, but this has not resolved the issue.
Since the DNG files display correctly in the app and the problem occurs only in the web interface, I suspect a bug in Immich's web rendering rather than an issue with the image files themselves.
I had previously opened a separate thread regarding this issue. However, it was closed, even though I believe this involves a different bug. My original post focused specifically on the discrepancy in how DNG files are displayed in the app versus the web browser—a point that was unfortunately not addressed.
Can anyone else reproduce or confirm this? Is there a known bug or a possible cause?

**Comment by mjh2901** [comment]
Immich 3.0.2 the errors loading certain DNG files (iphone) is pressent on both the iphone client and the web interface across OS's and browsers.

**Comment by Merlin-2504** [comment]
> Bei Immich 3.0.2 treten Fehler beim Laden bestimmter DNG-Dateien (iPhone) sowohl auf dem iPhone-Client als auch auf der Weboberfläche über verschiedene Betriebssysteme und Browser hinweg auf.

Unfortunately, these errors persist with DNG files captured on the iPhone 17 Pro. However, the issue is difficult to reproduce, as the images sometimes display correctly and other times do not; the behavior therefore occurs only sporadically.

**Comment by mjh2901** [comment]
I found the difference is did you edit the image in the photos app prior to upload which has ios generating a preview that does workSent from my iPhoneOn Jul 11, 2026, at 10:59 PM, Merlin-2504 ***@***.***> wrote:﻿Merlin-2504 left a comment (immich-app/immich#13029)

Bei Immich 3.0.2 treten Fehler beim Laden bestimmter DNG-Dateien (iPhone) sowohl auf dem iPhone-Client als auch auf der Weboberfläche über verschiedene Betriebssysteme und Browser hinweg auf.

Unfortunately, these errors persist with DNG files captured on the iPhone 17 Pro. However, the issue is difficult to reproduce, as the images sometimes display correctly and other times do not; the behavior therefore occurs only sporadically.

**Comment by Merlin-2504** [comment]
> Ich habe herausgefunden, dass der Unterschied darin besteht, ob Sie das Bild vor dem Hochladen in der Fotos-App bearbeitet haben. iOS generiert nämlich eine funktionierende Vorschau.***@***.***
I’d like to quickly clarify this to ensure I understand the statement correctly.

The affected images are unprocessed DNG files taken directly with an iPhone 17 Pro. Does the error occur with edited images, or also with unprocessed original shots? My files are not edited on the iPhone after capture.

**Comment by PineappleEmperor** [comment]
Adding my 2p. Also using an iphone 17 pro (max) - the ones that break for me seem to be the ones edited/rotated using the iphone photos app. Other unedited dng files seem to be fine.

**Comment by zxcOvosh** [comment]
support. version 3.0.2, photos.DNGs are not displayed in the browser or in the app either if you delete them locally.

**Comment by timbatog** [comment]
I attempted to deploy this today and all builds fine, however, when checking the status it seems to alternate between the Adobe and regular ,stock, lib raw.

Rechecking 'status' shows a different answer (Adobe/stock lib raw) every few minutes. Any one else seeing this?

**Comment by PseudoResonance** [comment]
> I attempted to deploy this today and all builds fine, however, when checking the status it seems to alternate between the Adobe and regular ,stock, lib raw.

Deploy what? There's a bunch of different options people have come up with.

**Comment by psenos** [comment]
Same issue with RAW files (DNG) files on Samsung Galaxy S26 Ultra on v3.0.3. Thank you for the fix on Linux Docker

**Comment by Merlin-2504** [comment]
The DNG files from my iPhone 17 Pro now work with the app and in the web browser. I'm using Server 3.0.2 and iOS App 3.0.3.
The DNG files are unedited.

**Comment by Legenda374** [comment]
I encountered a problem where thumbnails are not generated for my DNG photos from the iPhone 15 Pro that were edited through the iPhone gallery. That's why I get an error when viewing these photos through the browser and through the application. I'm used Server 3.0.1

```
[Nest] 6  - 07/27/2026, 7:34:21 PM   ERROR [Microservices:{"id":"xxxxxxxxx"}] Unable to run job handler (AssetGenerateThumbnails): Error: Input file has corrupt header: dcrawload: unable to read: Unsupported file format or not RAW file
Error: Input file has corrupt header: dcrawload: unable to read: Unsupported file format or not RAW file
    at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/usr/src/app/server/dist/repositories/media.repository.js:110:68)
    at MediaService.decodeImage (/usr/src/app/server/dist/services/media.service.js:185:59)
    at async MediaService.extractOriginalImage (/usr/src/app/server/dist/services/media.service.js:196:44)
    at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:212:32)
    at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:152:25)
    at async JobService.onJobRun (/usr/src/app/server/dist/services/job.service.js:78:30)
    at async EventRepository.onEvent (/usr/src/app/server/dist/repositories/event.repository.js:91:13)
    at async /usr/src/app/server/node_modules/.pnpm/bullmq@5.78.1/node_modules/bullmq/dist/cjs/classes/worker.js:589:32
```

But while I was writing this message my server was updated to 3.0.3 it seemed to fix the issue after update and regenerating the thumbnails.
It looks like the problem in 3.0.3 has been solved, you just need to update your server.

**Comment by pax0707** [comment]
I just did an import on 3.1 and no thumbnails still.
iOS, JPEG XL.

`2026-08-03T20:30:44.923603785Z [Nest] 7  - 08/03/2026, 10:30:44 PM   ERROR [Microservices:{"id":"fe513189-1a8f-469a-8e5c-e979302510ae"}] Unable to run job handler (AssetGenerateThumbnails): Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
2026-08-03T20:30:44.925158197Z Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
2026-08-03T20:30:44.925250670Z     at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/output.js:163:17)
2026-08-03T20:30:44.925262066Z     at MediaRepository.decodeImage (/usr/src/app/server/dist/repositories/media.repository.js:109:68)
2026-08-03T20:30:44.925266967Z     at MediaService.decodeImage (/usr/src/app/server/dist/services/media.service.js:188:59)
2026-08-03T20:30:44.925271513Z     at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
2026-08-03T20:30:44.925275698Z     at async MediaService.extractOriginalImage (/usr/src/app/server/dist/services/media.service.js:199:44)
2026-08-03T20:30:44.925280238Z     at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:215:32)
2026-08-03T20:30:44.925284696Z     at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:155:25)
2026-08-03T20:30:44.925289252Z     at async JobService.onJobRun (/usr/src/app/server/dist/services/job.service.js:78:30)
2026-08-03T20:30:44.925293569Z     at async EventRepository.onEvent (/usr/src/app/server/dist/repositories/event.repository.js:91:13)
2026-08-03T20:30:44.925300279Z     at async /usr/src/app/server/node_modules/.pnpm/bullmq@5.80.5/node_modules/bullmq/dist/cjs/classes/worker.js:589:32`

**Comment by anonanonis** [comment]
I have the same issue using an **iPhone 17 Pro Max**, RAW DNG photos are showing the same _"error loading image"_ message. The strangest part is that when I use my old **iPhone 14 Pro Max**, RAW DNG photos are showing correctly. Can I dev please respond? There are a dozen users on /r/immich reporting the same issue.

**Comment by Oshyan** [comment]
I think the likely cause of all the newer iPhone reports is the same as the original here. If that's the case, the important thing about that is: this is **not** something Immich devs can directly fix. They can modify Immich to make it easier for _other_ people to fix it (upstream library authors, people self-modifying like me), but they fundamentally depend on external libraries to decode raw images of this kind. It is _those_ external libraries that need to be updated, or Immich needs to be extended to work with Adobe DNG as I've suggested above.

In the meantime someone with the iPhone DNG issue should try the patch solution. I think it will either fix the problem, or if not, it will suggest that it might be better to open it as a separate issue report to track. I encourage someone to try this:
https://github.com/Oshyan/immich-dng-extended
If you've already setup self-hosting for Immich, it should be fairly easy to do this, and it's immediately reversible if it doesn't work or anything goes wrong.

**Comment by Lovos98** [comment]
Enabling **Prefer embedded preview** (Administration → Settings → Image Settings) fixes this for a large class of the files in this thread, and I have not seen it mentioned here.

My library, Immich v3.1.0, iPhone 16 Pro ProRAW:

```
DNG assets with a thumbnail, before: 36 / 2902
DNG assets with a thumbnail, after:  2902 / 2902     0 failures
```

The files are `Compression: JPEG XL` with a 4032x3024 embedded preview. Reproducing the failure directly through Immich's own pipeline:

```
sharp("IMG_7694.DNG").resize(1440).jpeg().toBuffer()
dcrawload: unable to unpack: Unsupported file format or not RAW file
```

With the setting on, `extractOriginalImage` feeds the embedded preview to the decoder as a buffer and libraw is never involved.

One thing worth knowing, because it explains why some people report the setting not helping. `shouldUseExtractedImage` only accepts the preview when `min(width, height) >= image.preview.size`, which defaults to 1440. A preview smaller than that is discarded and you are back to the failing path with no explanation.

This is not a replacement for real DNG SDK support. It is the camera's own render rather than a raw decode, and it does nothing for RAW files that carry no embedded preview at all. But for iPhone and Samsung ProRAW it turns a broken library into a working one with one toggle.

**Comment by patched-and-caffeinated** [comment]
The issue still exists on the DNG picture I upload to immich from my Samsung Galaxy S25 ultra, taken from the Expert RAW app. Turning on embedding does load the thumbnails, but in a pretty bad way with washed out colors. 
In the attached screenshot, left is the picture opened in Windows 11 photos app, while on the right is the picture displayed in immich (also looks the same in the photo library)

<img width="1843" height="763" alt="Image" src="https://github.com/user-attachments/assets/b728aab9-212d-4f29-a11b-616e0bf31f13" />

**Comment by Oshyan** [comment]
> The issue still exists on the DNG picture I upload to immich from my Samsung Galaxy S25 ultra, taken from the Expert RAW app. 

As far as I know there is no expectation that the issue has been fixed. The changelog does not reflect any fixes aimed at these issues:
https://github.com/immich-app/immich/releases

I can only repeat what I've suggested before: try the external Adobe DNG method described above and in the linked Github repo:
https://github.com/Oshyan/immich-dng-extended

**Comment by PseudoResonance** [comment]
Due to licensing issues, it's almost 100% certain this will never be fixed.

The only options are likely to patch your installation locally, or if Immich adds some kind of plugin system, to install a community plugin for support.

Saying it doesn't work will never change anything because sadly there is nothing that can be done to suddenly make it work.

**Comment by mjh2901** [comment]
The latest update has fixed my issues

**Comment by myusuf3** [comment]
Another root cause under this umbrella, on **v3.1.0**: Apple ProRAW DNGs using **JPEG XL** compression. 152 assets, all failing thumbnail generation with:

```
Unable to run job handler (AssetGenerateThumbnails):
  Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
    at Sharp.toBuffer (sharp@0.34.5/lib/output.js:163:17)
    at MediaRepository.decodeImage (media.repository.js:109:68)
```

`exiftool` on one of them:

```
DNG Version               : 1.7.0.0
Compression               : Unknown (52546)     <- JPEG XL
Photometric Interpretation: Linear Raw
Bits Per Sample           : 10 10 10
Unique Camera Model       : iPhone17,2 back ultra wide camera
Preview Image Length      : 6364242
```

libvips parses the container but can't unpack the pixel data, and ImageMagick fails identically since it's the same libraw underneath:

```
vipsheader IMG.dng   -> 8064x6048 uchar, 3 bands, srgb, dcrawload    # header fine
vips copy IMG.dng x.jpg -> dcrawload: unable to unpack
magick IMG.dng -resize 300x -> Unsupported file format or not RAW file
                               @ error/dng.c/ReadDNGImage/536
```

**Not the same as #29805.** Magic bytes are `4d 4d 00 2a` + `APPLEDNG` — a genuine big-endian TIFF, not one of the JPEGs-with-a-.dng-extension from Apple's upload path. The rename-to-`.jpg` workaround from that thread would be wrong for these.

**Workaround:** enabling **Prefer embedded preview** (`image.extractEmbedded: true`) fixed all 152 — 0 failures, ~90 s. These files carry a full-size embedded JPEG (6.4 MB above), and the resulting previews are full quality, not degraded.

Worth noting #28870 proposed exactly this as an automatic fallback on decode failure. It was auto-closed by the template bot the day it opened and never reviewed — might be worth reopening, since it would cover both this and the Lightroom-stitched files in the original report.

**Comment by JefNewTech** [comment]
Can confirm this is still happening on **v3.1.0** (`ghcr.io/immich-app/immich-server:v3.1.0`), also on an **iPhone 17 Pro Max**, TrueNAS SCALE deployment.

Thumbnail generation fails on every single ProRAW DNG synced from this device — not intermittent, 100% failure rate so far, confirmed on a batch of assets uploaded together plus one uploaded separately a couple hours later.

Server logs (same error on every asset, only asset ID changes):

```
[Nest] 7  - 08/26/2026, 6:14:20 PM   ERROR [Microservices:{"source":"upload","id":"169a91ef-6df7-4f98-b1c5-f0c8b134c299"}] Unable to run job handler (AssetGenerateThumbnails): Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
    at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/usr/src/app/server/dist/repositories/media.repository.js:109:68)
    at MediaService.decodeImage (/usr/src/app/server/dist/services/media.service.js:188:59)
    at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
    at async MediaService.extractOriginalImage (/usr/src/app/server/dist/services/media.service.js:199:44)
    at async MediaService.generateImageThumbnails (/usr/src/app/server/dist/services/media.service.js:215:32)
    at async MediaService.handleGenerateThumbnails (/usr/src/app/server/dist/services/media.service.js:155:25)
    at async JobService.onJobRun (/usr/src/app/server/dist/services/job.service.js:78:30)
    at async EventRepository.onEvent (/usr/src/app/server/dist/repositories/event.repository.js:91:13)
    at async /usr/src/app/server/node_modules/.pnpm/bullmq@5.80.5/node_modules/bullmq/dist/cjs/classes/worker.js:589:32

[Nest] 7  - 08/26/2026, 6:14:21 PM   ERROR [Microservices:{"source":"upload","id":"cafad507-3dac-4656-ac76-7a2fab0db348"}] Unable to run job handler (AssetGenerateThumbnails): Error: dcrawload: unable to unpack: Unsupported file format or not RAW file

[Nest] 7  - 08/26/2026, 8:47:28 PM   ERROR [Microservices:{"id":"d395bec5-ce8e-4d82-98ed-28e6da86330a"}] Unable to run job handler (AssetGenerateThumbnails): Error: dcrawload: unable to unpack: Unsupported file format or not RAW file
```

10+ distinct asset IDs hit this exact same error in the logs; trimmed to a representative sample above.

Worth noting: this error path is `dcrawload`, whereas older DNG-related reports (e.g. #13029, #13733) show `magickload: Magick: Unsupported file format or not RAW file`. Might indicate the 17 Pro Max's DNG structure is hitting a different decode path than previous iPhone ProRAW files, which could be relevant for narrowing down the root cause.

"Prefer embedded preview" doesn't help either.

**Comment by drunkrhin0** [comment]
> Enabling **Prefer embedded preview** (Administration → Settings → Image Settings) fixes this for a large class of the files in this thread, and I have not seen it mentioned here.
> 
> My library, Immich v3.1.0, iPhone 16 Pro ProRAW:
> 
> ```
> DNG assets with a thumbnail, before: 36 / 2902
> DNG assets with a thumbnail, after:  2902 / 2902     0 failures
> ```
> 
> The files are `Compression: JPEG XL` with a 4032x3024 embedded preview. Reproducing the failure directly through Immich's own pipeline:
> 
> ```
> sharp("IMG_7694.DNG").resize(1440).jpeg().toBuffer()
> dcrawload: unable to unpack: Unsupported file format or not RAW file
> ```
> 
> With the setting on, `extractOriginalImage` feeds the embedded preview to the decoder as a buffer and libraw is never involved.
> 
> One thing worth knowing, because it explains why some people report the setting not helping. `shouldUseExtractedImage` only accepts the preview when `min(width, height) >= image.preview.size`, which defaults to 1440. A preview smaller than that is discarded and you are back to the failing path with no explanation.
> 
> This is not a replacement for real DNG SDK support. It is the camera's own render rather than a raw decode, and it does nothing for RAW files that carry no embedded preview at all. But for iPhone and Samsung ProRAW it turns a broken library into a working one with one toggle.

Enabling embedded preview seems to have mostlyh fixed this for me. Still looking into the dng-extended repo

**Comment by DeIphicOracIe** [comment]
Adding a reproducible data point for why "Prefer embedded preview" fails on **DxO PureRAW 6 linear DNGs** (v3.1.0 on Docker), connecting to the failures noted by @JefNewTech and @Lovos98.



### The Problem

With "Prefer embedded preview" enabled, thumbnail generation crashes during preview validation:

1. `Error: Input buffer contains unsupported image format` (at `Sharp.metadata` inside `MediaService.shouldUseExtractedImage`)

2. Fallback to full RAW decode: `Error: dcrawload: unable to unpack: Unsupported file format or not RAW file`



### Root Cause

Before `shouldUseExtractedImage` can evaluate dimensions (`min(w, h) >= 1440`), Sharp rejects the buffer returned by `extractOriginalImage`. Immich's preview extraction logic grabs a malformed stream (such as uncompressed thumbnail bytes or raw IFD offsets) rather than the actual JPEG stream.



However, scanning the 7.94 MB DNG for JPEG SOI markers (`0xFFD8FF`) shows that the file **does contain a valid, full-resolution embedded JPEG** inside its SubIFD:

- Offset `557856` (length: ~2.29 MB)



Testing this slice directly inside the `immich_server` container:

```bash

node -e '

const fs = require("fs");

const sharp = require("/usr/src/app/server/node_modules/sharp");

const buf = fs.readFileSync("/path/to/sample.dng");



// Slicing the SubIFD JPEG stream found at offset 557856

const jpegBuf = buf.subarray(557856, 557856 + 2347010);



sharp(jpegBuf).metadata()

  .then(m => console.log(`Decoded successfully: ${m.width}x${m.height} ${m.format}`))

  .catch(err => console.error("Sharp error:", err.message));

'

# Output: Decoded successfully: 5184x3888 jpeg

```



The embedded preview is an intact 20 MP JPEG. If `extractOriginalImage` correctly located and sliced this SubIFD preview stream instead of passing a malformed buffer to Sharp, these files wouldn't hit the `shouldUseExtractedImage` failure and wouldn't need to touch `dcrawload` at all.
